### PR TITLE
feat(build): delete gulp v3 and replace by scripts tasks

### PR DESCRIPTION
**Pull request by samuel-gomez-axa**
#621 : All builded css for packages are generated by Gulp tasks but the version of Gulp is not compatible with node 12
### Person(s) for reviewing proposed changes : @guillaumechervetaxa 

**Solution** : replace all tasks with vanilla scripts and remove Gulp v3

- delete gulpfile.js
- delete Gulp tasks from ./scripts/tasks/*
- create style.js in ./scripts that replace all gulp tasks
- create clean.js in ./scripts for clean all ./dist packages

I also **deleted** several tasks : 
- bootstrap-theme (boostrap bundle)
- split-bootstrap-component-universes (splitted bootstrap components themed by universes)
- split-bootstrap-component (splitted bootstrap components)
- universe-bundle (generate af-toolkit-core themed by universes)
- svgstore (generation of svg sprite with icons) <sup style="color:red">(1)</sup>
- iconfont (generation of icon font glyphicon) <sup style="color:red">(1)</sup>
- clean (delete all /dist packages)
- cleanlock (delete package-lock)
- generatelock (generate package-lock)

<sup style="color:red">(1)</sup> Theses tasks generated the icon font files and the sprite for icons.
image
but i would like to create icons React Components instead in next PR.
So i copy theses generates files in legacy directory sources and just copy it in /dist for non breaking changes
image

The Design System Website has already updated like SVG React components .

Finally, i deleted the storybook/styles because its unused.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -161,7 +147,13 @@
     "watch": "1.0.2"
   },
   "dependencies": {
+    "autoprefixer": "^9.8.0",
+    "clean-css": "^4.2.3",
+    "cuid": "^2.1.4",
```
Comment (xballoy) [inline package.json:152]: Do we use cuid ?

File: package.json [hunk 2]
```diff
@@ -161,7 +147,13 @@
...
+    "find": "^0.3.0",
     "glob-stream": "^6.1.0",
+    "moment": "~2.22.2",
```
Comment (xballoy) [inline package.json:155]: We can remove `moment` from package.json, it's not used in scripts.

File: package.json [hunk 3]
```diff
@@ -170,7 +162,8 @@
     "react-popper": "^2.2.3",
     "@popperjs/core": "^2.4.0",
     "react-select": "^3.1.0",
-    "react-transition-group": "~2.3.1"
+    "react-transition-group": "~2.3.1",
```
Comment (xballoy) [inline package.json:165]: We can remove `react-transition-group` from package.json, it's not used in scripts.

**Comment by xballoy** [comment]
Can we fix this warning when we run `style.js`:
```
Build style packages   Starting 
Without `from` option PostCSS could generate wrong source map and will not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning.
Build style packages   Finished 
```

File: package.json (hunk 2 above)
Comment (xballoy) [reply package.json:155]: Maybe we can do that in another PR to clean the dependencies

File: package.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
   },
   "scripts": {
     "publish": "lerna publish",
-    "postinstall": "npm run clean && lerna bootstrap && npm run style",
+    "postinstall": "npm run clean && lerna bootstrap --no-ci && npm run style",
```
Comment (xballoy) [inline package.json:11]: Pourquoi tu as besoin de faire un `--no-ci` ?

File: package.json (hunk 2 above)
Comment (samuel-gomez-axa) [reply package.json:155]: I'm agree to clean all unused packages in another PR

File: package.json (hunk 4 above)
Comment (samuel-gomez-axa) [reply package.json:11]: C'est parce que de base le lerna bootstrap est en CI : https://github.com/lerna/lerna/tree/master/commands/bootstrap#--no-ci
Comme nous avons supprimé les package-lock des packages, on ne peut plus faire de npm ci

File: packages/Form/Input/date/src/datepicker.scss [hunk 5]
```diff
@@ -15,6 +15,7 @@ $border-radius-width: 3px;
     top: 0.7rem;
     color: $color-dusty-gray;
     z-index: 2;
+    width: 17px;
```
Comment (arnaudforaison) [inline packages/Form/Input/date/src/datepicker.scss:18]: Cet ajout ne va rien casser ? J'ai cru comprendre que le datepicker était sensible ;-)

**Comment by samuel-gomez-axa** [comment]
@xballoy , i fixed the postcss warning and fix the browserlist warning.

File: packages/Form/Input/date/src/datepicker.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (samuel-gomez-axa) [reply packages/Form/Input/date/src/datepicker.scss:18]: oui, c'est surtout le fait d'avoir supprimer les tasks pour la génération des font icons. J'ai anticipé le CSS qui permettra d'avoir des composants React SVG pour l'utilisation des icons avec le toolkit.

[the remark this answers is not in the document]
Comment (samuel-gomez-axa) [reply packages/Form/Input/date/src/datepicker.scss:18]: tout en faisant attention de ne pas avoir de breaking change avec ceux qui utilisent les font icons ;)

**Pull request merged**